flaggers/duplicate.py

```diff
@@ -27,8 +28,18 @@ def flag(self, row_id, data):
         # of which are present within the Portal's database.
         # TODO: This is a really slow way of flagging duplicates, we definitely
         # should come up with a different way.
-        if row_id in list(np.where(data.duplicated(keep=False))[0]):
-            flag.append(Flags.DUPLICATE)
+        #if row_id in list(np.where(data.duplicated(keep=False))[0]):
+        #if row_id in duplicate_list:
+            #flag.append(Flags.DUPLICATE)
+
+        #Above check doesn't work because data contains no row_id, and matches happen on any other values which match passed in int (row_id)
```

Comment:
I don't think this is the reason why the above check doesn't work. It doesn't rely on any row_id column to begin with. `row_id` is a misnomer, it should have been called row_index, my bad.

Reply:
Don't worry too hard about this method, I'm in the middle of gutting it for my issue next sprint.